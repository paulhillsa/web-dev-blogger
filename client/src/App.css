/* General */

* {
  margin: 0;
  padding: 0;
  align-items: center;
  justify-content: center;
}

p {
  font-size: 1.5rem;
  font-weight: lighter;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
}

li {
  list-style: disc;
  font-size: 1rem;
  font-weight: lighter;
  padding: 0.5rem;
}

.form-label {
  font-size: 1rem;
  font-weight: lighter;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
}

/* Landing Page */

#landing-page-h1 {
    font-size: 5rem;
    font-weight: bold;
    color: #007bff;
    text-align: center;
    transform: translatey(200px);
}

#landing-page-h2{
    font-size: 2rem;
    font-weight: lighter;
    color: rgb(0, 0, 0);
    text-align: center;
    transform: translatey(200px);
}

.landing-page-buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    width: 15%;
    transform: translatey(200px);
    transform: translateX(10%);
}

/* Blogs */

#big-blog-container{
  width: 100%;
}

.blogs-form-button {
  padding: 2%;
  display: flex;
  flex-direction: column;
}


#blog-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transform: translatex(2%);
  width: 80%;
  height: 100%;
  border-radius: 10px;
  box-shadow: 0px 0px 5px rgb(0, 0, 0);
  margin: 100px;
  margin-top: 0%;

}

#blog-content {
padding: 100px;
padding-top: 10px;
padding-bottom: 20px;
}

#blog-image {
  object-fit: cover;
  width: 100%;
  height: 200px;
  border-radius: 10px;
}

#web-dev-h1 {
    font-size: 4rem;
    font-weight: bold;
    color: rgb(0, 0, 0);
    text-align: center;
    padding: 4%;
}

#blogs-form-h1{
    font-size: 2rem;
    font-weight: lighter;
    color: rgb(0, 0, 0);
    text-align: center;
    padding: 1%;
}

#blogs-form-content{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 50%;
    padding: 1%;
}

.blogs-form-group{
 width: 100%;
}

.admin-control {
  padding: 2%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

#blog-title {
  font-size: 4rem;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
  padding: 2%;
}

#web-dev-h1 {
  font-size: 4rem;
  font-weight: bold;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
  padding: 5%;
}

#readmore-button{
  padding: 1%;
  margin: 1.5%;
  display: flex;
  flex-direction: column;
}

#edit-delete-button{
  padding: 1%;
  margin: 1.5%;
  display: flex;
  flex-direction: row;
}

#edit-button{
  padding: 10%;
  margin: 10%;
}

#delete-button{
  padding: 10%;
  margin: 10%;
}

/* Navbar */

.navbar {
  padding: 2%;
  width: 100%;
  height: 10%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.login-false-navbar {
  display: flex;
  flex-direction: row;
  align-items: left;
  padding: 1%;
}

.login-true-navbar {
  display: flex;
  flex-direction: row;
  align-items: left;
  padding: 1%;
}

.logout-button-navbar{
  display: flex;
  position: absolute;
  left: 90%;
  padding: 1%;
}

.login-button-navbar{
  display: flex;
  position: absolute;
  left: 90%;
  padding: 1%;
}

.register-button-navbar{
  display: flex;
  position: absolute;
  left: 84%;
  padding: 1%;
}

/* Login */

.login-container {
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: left;
  width: 50%;
  padding: 10%;
}

#login-text {
  font-size: 2rem;
  font-weight: lighter;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
}

.login-submit-button-container {
  text-align: right;
  justify-content: right;
  padding: 5%;
}

/* Register */

.register-container {
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: left;
  width: 50%;
  padding: 10%;
}

#register-text {
  font-size: 2rem;
  font-weight: lighter;
  color: rgb(0, 0, 0);
  text-align: left;
  justify-content: left;
}

.register-submit-button-container {
  text-align: right;
  justify-content: right;
  padding: 5%;
}
