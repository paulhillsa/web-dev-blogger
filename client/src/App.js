//App.js will be the main file for the client side
// it provides the context to all children components
// as well as render the router

import './App.css';
import React from 'react';
import Router from './Router';
import axios from 'axios';
import {AuthContextProvider} from './context/AuthContext';

axios.defaults.withCredentials = true;

function App() {
  return <>
    {/* Import react-bootstrap */}
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css"
      integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
      crossorigin="anonymous"
    />
    {/* Provide Context to all routes */}
    <AuthContextProvider>
      <Router />
    </AuthContextProvider>
    </>;
}


export default App;
