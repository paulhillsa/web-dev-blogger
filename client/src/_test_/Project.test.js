import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import axios from 'axios';

//Snapshot test for the App component
test('App renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});


// unit test for the axios call
describe('axios call', () => {
    test('should return a promise', () => {
        expect(axios.get).toBeDefined();
    });
}
);