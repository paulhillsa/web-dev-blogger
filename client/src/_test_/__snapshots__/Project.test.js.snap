// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <link
    crossorigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css"
    integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
    rel="stylesheet"
  />,
  <div
    className="navbar nav"
    onKeyDown={[Function]}
  />,
  <div />,
]
`;
